Emby.Server.Implementations/Syncplay/SyncplayController.cs

```diff
@@ -228,190 +271,271 @@ public void SessionJoin(SessionInfo session, JoinGroupRequest request)
-                            );
-                            var command = NewSyncplayCommand(SendCommandType.Play);
-                            SendCommand(session, BroadcastType.AllExceptSession, command);
-                        }
-                        else
-                        {
-                            // Client, that was buffering, resumed playback but did not update others in time
-                            delay = _group.GetHighestPing() * 2;
-                            delay = delay < _group.DefaulPing ? _group.DefaulPing : delay;
-
-                            _group.LastActivity = currentTime.AddMilliseconds(
-                                delay
-                            );
-
-                            var command = NewSyncplayCommand(SendCommandType.Play);
-                            SendCommand(session, BroadcastType.AllGroup, command);
-                        }
-                    }                    
-                }
-                else
-                {
-                    // Make sure client has latest group state
-                    var command = NewSyncplayCommand(SendCommandType.Play);
-                    SendCommand(session, BroadcastType.SingleSession, command);
+                        // Client, that was buffering, resumed playback but did not update others in time
+                        delay = _group.GetHighestPing() * 2;
+                        delay = delay < _group.DefaulPing ? _group.DefaulPing : delay;
+
+                        _group.LastActivity = currentTime.AddMilliseconds(
+                            delay
+                        );
+
+                        var command = NewSyncplayCommand(SendCommandType.Play);
+                        SendCommand(session, BroadcastType.AllGroup, command, CancellationToken.None);
+                    }
                 }
             }
-            else if (request.Type.Equals(PlaybackRequestType.UpdatePing))
+            else
             {
-                _group.UpdatePing(session, request.Ping ??= _group.DefaulPing);
+                // Group was not waiting, make sure client has latest state
+                var command = NewSyncplayCommand(SendCommandType.Play);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
             }
         }
 
+        /// <summary>
+        /// Sanitizes the PositionTicks, considers the current playing item when available.
+        /// </summary>
+        /// <param name="positionTicks">The PositionTicks.</param>
+        /// <value>The sanitized PositionTicks.</value>
+        private long SanitizePositionTicks(long? positionTicks)
+        {
+            var ticks = positionTicks ??= 0;
+            ticks = ticks >= 0 ? ticks : 0;
+            if (_group.PlayingItem.RunTimeTicks != null)
+            {
+                var runTimeTicks = _group.PlayingItem.RunTimeTicks ??= 0;
+                ticks = ticks > runTimeTicks ? runTimeTicks : ticks;
```

Comment:
```suggestion
                ticks = ticks > _group.PlayingItem.RunTimeTicks ? _group.PlayingItem.RunTimeTicks : ticks;
```
you've already checked that `_group.PlayingItem.RunTimeTicks` is not `null`, why check again?..

Reply:
The first check is for `null`, the other ensures that `ticks` never go beyond `_group.PlayingItem.RunTimeTicks`. It looked more clear to me :sweat_smile:.

The `??=` needs to be changed though, as you pointed out bellow.

Reply:
My bad, that's indeed checking twice the same thing.